File: Java/274. H-Index.java
class Solution {
    public int hIndex(int[] citations) {
        int[] counts = new int[1001];
        for (int c: citations) {
            counts[c]++;
        }

        for (int i = 1000; i > 0; i--) {
            if (counts[i] >= i) {
                return i;
            }
            counts[i - 1] += counts[i];
        }

        return 0;
    }
}

File: Python3/274. H-Index.py
class Solution:
    def hIndex(self, citations: List[int]) -> int:
        total = len(citations)
        counts = [0] * (total + 1)

        for c in citations:
            counts[min(c, total)] += 1

        for i in range(total, 0, -1):
            if counts[i] >= i:
                return i

            counts[i - 1] += counts[i]

        return 0